### PR TITLE
Paged search ordering

### PR DESCRIPTION
**Pull request by AliveMen**
add thenById

### CODE REVIEW

File: VirtoCommerce.CatalogModule.Data/Services/CatalogSearchServiceImpl.cs [hunk 1]
```diff
@@ -133,7 +131,7 @@ protected virtual void SearchCategories(SearchCriteria criteria, SearchResult re
                 //Try to replace sorting columns names
                 TryTransformSortingInfoColumnNames(_categorySortingAliases, sortInfos);
 
-                query = query.OrderBySortInfos(sortInfos);
+                query = query.OrderBySortInfos(sortInfos).ThenBy(q => q.Id);
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Data/Services/CatalogSearchServiceImpl.cs]: According to naming style need to use x for such linq with one argument (in other cases too).

File: VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace VirtoCommerce.CatalogModule.Data.Extensions
+{
+    public static class SortInfoExtension
+    {
+        public static SortInfo[] EnsureIdIncluded(this SortInfo[] sortInfos)
+        {
+            if (!sortInfos.IsNullOrEmpty())
+            {
+                if (!sortInfos.Any(x => x.SortColumn.EqualsInvariant("id")))
```
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs]: Move to the previous if()

File: VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+                {
+                    var sortInfoLst = sortInfos.ToList();
```
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs]: There is no such word as 'Lst'

File: VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+                    sortInfoLst.Add(new SortInfo()
+                    {
+                        SortColumn = "id"
+                    });
+                    return sortInfoLst.ToArray();
```
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs]: Single return

File: VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs [hunk 5]
```diff
@@ -7,19 +8,16 @@ public static class SortInfoExtension
     {
         public static SortInfo[] EnsureIdIncluded(this SortInfo[] sortInfos)
         {
-            if (!sortInfos.IsNullOrEmpty())
+            var sortInfoList = sortInfos?.ToList() ?? new List<SortInfo>();
```
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs]: Don't create new list if original sortInfos array already contains 'id'

Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs]: I'm create new list when original sortInfos == null only

Comment (artem-dudarev) [reply VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs]: `ToList()` creates new list

Comment (yecli) [reply VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs]: Probably Artem meant we could change the result under condition
`if (!sortInfoList?.Any(x => x.SortColumn.EqualsInvariant("id")))`

Comment (artem-dudarev) [reply VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs]: I mean there is no need to create new list and then convert it back to array if we already have 'id' in original array.

File: VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+using System.Linq;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.CatalogModule.Data.Extensions
+{
+    public static class SortInfoExtension
+    {
+        public static SortInfo[] EnsureIdIncluded(this SortInfo[] sortInfos)
```
Comment (artem-dudarev) [inline VirtoCommerce.CatalogModule.Data/Extensions/SortInfoExtension.cs]:
```csharp
public static SortInfo[] EnsureIdIncluded(this SortInfo[] sortInfos)
{
    var result = sortInfos;

    if (sortInfos == null || !sortInfos.Any(x => x.SortColumn.EqualsInvariant("id")))
    {
        var list = sortInfos?.ToList() ?? new List<SortInfo>();
        list.Add(new SortInfo { SortColumn = "id" });
        result = list.ToArray();
    }

    return result;
}
```

**Comment by yecli** [comment]
@tatarincev I think we could use that extensions to platform and use this `OrderBySortInfosWithId` method as generic solution.

**Pull request merged**